CMSgov/mint-app
# [EASI-4394] Models by Solution homepage

# EASI-4394

## Description


- Updated `GlobalTableFilter` to not have decoupled state with query.  State setter and state are now coupled.  This fixes issues with discrepancies between text in search box and state of global filter.
- Component for banner with button group for toggling status
  - Statuses groups should match comment per Figma (planned is anything not active or ended, etc)
- Card component for rendering models of solutions
- Parent/wrapper component to house all individual components
- Mobile/Tablet responsive
  - Should render select dropdown on tablet or when too many solutions
- Unit tests


## How to test this change


Note: When searching with query and no results, query box and results will persist across status buttons, even though search is normally not available on statuses that have less than 4 results.  This is to persists the search while notifying the user the search input is still active.

- Seed db
- Configure homepage settings to include models by solution
- Add a few solutions
- Add more than 4 models to a single solution
- Verify search and pagination appear and work correctly (pagination appears at more than 3)
- Test mobile responsiveness

## PR Author Checklist




## PR Reviewer Guidelines

- It's best to pull the branch locally and test it, rather than just looking at the code online!
- When approving a PR, provide a reason _why_ you're approving it
  - e.g. "Approving because I tested it locally and all functionality works as expected"
  - e.g. "Approving because the change is simple and matches the Figma design"
- Don't be afraid to leave comments or ask questions, especially if you don't understand why something was done! (This is often a great time to suggest code comments or documentation updates)
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
